### PR TITLE
fix: execute commands relative to wrapper

### PR DESCRIPTION
**Pull request by lili2311**
- [ ] Ready for review
- [ ] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Execute the commands relative to the wrapper to fix a bug with `--all-projects` where if the command is executed in a dir above a wrapper then gradle fails to execute even if we successfully locate the wrapper.

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -646,25 +659,38 @@ function toCamelCase(input: string) {
...
   pathToWrapper = path.resolve(root, wrapperScript);
   if (fs.existsSync(pathToWrapper)) {
-    return quotLocal + pathToWrapper + quotLocal;
+    return {
+      command: quotLocal + pathToWrapper + quotLocal,
+      pathToWrapper: path.parse(pathToWrapper).dir,
+    };
   }
-  return 'gradle';
+  return { command: 'gradle', pathToWrapper: root };
```
Comment (aviadhahami) [inline lib/index.ts:693]: why not `pathToWrapper`?

File: lib/index.ts [hunk 2]
```diff
@@ -524,9 +531,15 @@ async function getAllDeps(
...
       const { snykGraph, projectVersion } = extractedJson.projects[projectId];
 
+      // TODO: this should be relative to wrapper
+      // can't be changed without causing naming mismatch to existing projects
+      // may need a migration
       let projectName = path.basename(root);
 
       if (projectId !== extractedJson.defaultProject) {
+        // TODO: this should be relative to wrapper
+        // can't be changed without causing naming mismatch to existing projects
```
Comment (aviadhahami) [inline lib/index.ts:541]: how can we handle this?

Comment (lili2311) [reply lib/index.ts:544]: let me test locally to confirm this is definitely going to be a problem or not.

Comment (lili2311) [reply lib/index.ts:544]: Doesn't seem to affect the projectID 🎉 Please can you also verify?

File: lib/index.ts (hunk 1 above)
Comment (lili2311) [reply lib/index.ts]: to keep it equivalent to before. Returning this triggers some other logic when a wrapper is found

**Pull request closed without merging**